runtime/shared_common/ClasspathManagerImpl2.cpp

```diff
@@ -1099,7 +1114,26 @@ SH_ClasspathManagerImpl2::validate(J9VMThread* currentThread, ROMClassWrapper* f
 	if (localFoundAtIndex == -1) {
 		/* We didn't find an exact classpath match. Walk the classpaths and check for validity. Note that the timestamp check is incorporated for efficiency. */
 		localFoundAtIndex = localValidate_CheckAndTimestampManually(currentThread, testCP, testCPIndex, compareTo, foundIdentified, &addToIdentified, staleItem);
+		/* If the classpath has just been identified by localValidate_CheckAndTimestampManually, store it in the identified array */
+		if (addToIdentified && (compareTo->getType()==CP_TYPE_CLASSPATH) && (*_runtimeFlagsPtr & J9SHR_RUNTIMEFLAG_ENABLE_LOCAL_CACHEING)) {
+			/* Identification succeeded. Store identified classpath. */
+			if (local_StoreIdentified(currentThread, compareTo, testCP)==-1) {
+				/* trace event is at level 1 and trace exit message is at level 2 as per CMVC 155318/157683  */
+				Trc_SHR_CMI_validate_ExitError_Event(currentThread, foundROMClass, compareTo, confirmedEntries);
+				Trc_SHR_CMI_validate_ExitError(currentThread);
+				if (releaseIdentifiedMutex) {
```

Comment:
Seems if releaseIdentifiedMutex is false, the assert in local_StoreIdentified() would occur before getting here.

Reply:
Fixed.